infer/models/c/src/libc_basic.c

```diff
@@ -643,15 +643,27 @@ int fputc(int c, FILE* stream) {
 char* getcwd(char* buffer, size_t size) {
   int n;
   int size_buf;
-  n = __infer_nondet_int();
-
-  if (n > 0) {
-    __require_allocated_array(buffer);
-    size_buf = __get_array_length(buffer);
-    INFER_EXCLUDE_CONDITION(size > size_buf);
-    return buffer;
-  } else
-    return NULL;
+  char *result;
+
+  if (NULL == buffer) {
```

Comment:
Don't you mean `buffer != NULL`? I'm editing this PR and adding a test since you sent it so long ago (sorry about the delay, I didn't realise the CLA got accepted).